src/Uno.UI.Runtime.Skia.MacOS/MacOSWindowHost.cs

```diff
@@ -80,7 +82,15 @@ private void Draw(SKSurface surface)
 
 			if (RootElement?.Visual is { } rootVisual)
 			{
-				RootElement.XamlRoot?.Compositor.RenderRootVisual(surface, rootVisual, null);
+				int width = (int)nativeWidth;
+				int height = (int)nativeHeight;
+				var path = SkiaRenderHelper.RenderRootVisualAndReturnPath(width, height, rootVisual, surface);
+				if (path is { })
+				{
+					var svg = path.ToSvgPathData();
+					// remove first path since it cover the whole area
```

Comment:
@ramezgerges I'm not sure why but the first path returned covers everything, which clips all native elements. I removed it but the problem seems elsewhere.

Reply:
I just checked and `RenderRootVisualAndReturnPath` does indeed add the entire window viewport first, but it's intended. The idea is that we start with nothing clipped, and then we "chip away" at this full viewport whenever we find a native visual.

Notice the line with `mainPath.AddRect(new SKRect(0, 0, width, height));`
```csharp
			// the entire viewport
			var mainPath = new SKPath();
			mainPath.AddRect(new SKRect(0, 0, width, height));

			rootVisual.Compositor.RenderRootVisual(surface, rootVisual, (session, visual) =>
			{
				// minus the native-element areas
				if (visual.IsNativeHostVisual)
				{
					_clipPath.Reset();
					_clipPath.AddRect(canvas.DeviceClipBounds);
					_visualPath.Reset();
					_visualPath.AddRect(visual.TotalMatrix.ToMatrix3x2().Transform(new Rect(new Point(), visual.Size.ToSize())).ToSKRect());
					mainPath = mainPath.Op(_clipPath.Op(_visualPath, SKPathOp.Intersect), SKPathOp.Difference);
				}

				// plus the popups
				if (visual.IsPopupVisual)
				{
					_clipPath.Reset();
					_clipPath.AddRect(canvas.DeviceClipBounds); // possible QoL improvement: find a way to get the tight clip path (e.g. for rounded corners) not just the bounding rect
					_visualPath.Reset();
					_visualPath.AddRect(visual.TotalMatrix.ToMatrix3x2().Transform(new Rect(new Point(), visual.Size.ToSize())).ToSKRect());
					// note that popups are always traversed last in the tree, so there are no "races" between the paths of PopupVisuals and NativeHostVisuals
					mainPath = mainPath.Op(_clipPath.Op(_visualPath, SKPathOp.Intersect), SKPathOp.Union);
				}
			});
```

Reply:
Thanks for looking! I think (from the output I get) that the SVG path (made by Skia) is skipping the `Op` 😢

Reply:
This is what I get from `path.ToSvgPathData`:

```
M2048 0L0 0L0 1480L2048 1480L2048 0ZM776 240L613 240L613 255L776 255L776 240Z
```

Reply:
I'm not an SVG guru, but I think this is actually accurate! The thing to watch out for is the `FillType` (aka `fill-rule` in Web APIs). I've adjusted your string a tiny bit to `M2048 0L0 0L0 1480L2048 1480L2048 0ZM776 240L613 240L613 355L776 355L776 240Z` (I made the "native viewport" slightly bigger). The FillType is `EvenOdd` by default. Here's a snippet with your string. Notice that if you remove the `fill-rule` bit, you get a solid rectangle like you experience: https://www.svgviewer.dev/s/nK750twd

Reply:
good catch, the default is actually `kCAFillRuleNonZero`
and I'll need to reverse the shape (which was the original code)